{"uid":"1c0332ae3f13d75e","name":"test_turnlamp_synchronflash[8]","fullName":"tests.turnlamp.test_synchron_flash#test_turnlamp_synchronflash","historyId":"1c05a63287ebdb433768f434f6e1a2d9","time":{"start":1756272708549,"stop":1756272708549,"duration":0},"status":"broken","statusMessage":"openta.common.exceptions.PortError: Setup of Ports Failed: dSPACE.XIL.Testbench.ECUPort.Interfaces.Error.ECUPortException_impl: No valid license found. Vendor Description: CodeMeter License Server not found, Error 101.\n   at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CheckAndThrowException(ServiceException exception)\n   at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CreateInstance(InstanceType instanceType, Object[] args)\n   at dSPACE.XIL.Testbench.Client.Testbench.ECUPort.ECUPortFactory.CreateECUPort(String name)\n␤Failed to create 'ECU' DsEcuPort","statusTrace":"self = <openta.testing.testenv.TestEnvironmentAccess object at 0x7cdd73ac6c50>\n\n    def setup(self) -> None:\n        try:\n            with allure.step(\"Setup Ports\"):\n                _logger.info(\"Setup Ports\")\n                for port in Registry().get_ports():\n                    port_repr = f\"[{port.order}] '{port.name}' {port.__class__.__name__}\"\n    \n                    if port.state < PortState.CREATED <= port.target_state:\n                        _logger.info(f\"Setup {port_repr}: Create\")\n>                       port.create()\n\n.venv/lib/python3.11/site-packages/openta/testing/testenv.py:162: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <openta.ports.xilapi_dspace_ecu.DsEcuPort object at 0x7cdd72c90d50>\nkwargs = {}\next_int = <module 'dSPACE.XIL.Testbench.ExtendedInterfaces.Testbench'>\n\n    def create(self, **kwargs: typing.Any) -> None:  # noqa: ANN401, ARG002 # unknown keyword args by design\n        with log_port_lifecycle(_logger, self, \"create\"):\n            # call super create(), to create the testbench beforehand\n            super().create()\n    \n            # NOTE the assemblies have to be already registered\n            ext_int = importlib.import_module(\"dSPACE.XIL.Testbench.ExtendedInterfaces.Testbench\")\n>           self._port = ext_int.IDSTestbench(self._testbench).ECUPortFactory.CreateECUPort(self.name)\n                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nE           dSPACE.XIL.Testbench.ECUPort.Interfaces.Error.ECUPortException_impl: No valid license found. Vendor Description: CodeMeter License Server not found, Error 101.\nE              at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CheckAndThrowException(ServiceException exception)\nE              at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CreateInstance(InstanceType instanceType, Object[] args)\nE              at dSPACE.XIL.Testbench.Client.Testbench.ECUPort.ECUPortFactory.CreateECUPort(String name)\nE           Failed to create 'ECU' DsEcuPort\n\n.venv/lib/python3.11/site-packages/openta/ports/xilapi_dspace_ecu.py:100: ECUPortException_impl\n\nThe above exception was the direct cause of the following exception:\n\nrequest = <SubRequest 'ta' for <Function test_turnlamp_synchronflash[6]>>\n\n    @pytest.fixture(scope=\"session\")\n    def ta(request: pytest.FixtureRequest) -> typing.Generator[TestEnvironmentAccess, None, None]:\n        \"\"\"\n        Session scoped pytest fixture setting up the XIL API Framework.\n    \n        Requires either --openta-testenv-config or --openta-xilframework-config option.\n        \"\"\"\n    \n        if \"--collect-only\" not in sys.argv:\n            # Check if config options were provided when fixture is used in a test\n            testenv_config = request.config.getoption(\"--openta-testenv-config\")\n    \n            # but only one of them, they are mutual exclusive\n            if not testenv_config:\n                raise pytest.UsageError(\n                    \"The `ta` fixture requires `--openta-testenv-config` pytest option.\",\n                )\n    \n            # Read the test environment configuration\n            reader = TestEnvironmentConfigReader(testenv_config)  # pyright: ignore[reportArgumentType] : if it is not None, it is str\n            reader.add_to_registry()\n    \n            # and instantiate the TestEnvironmentAccess, which is returned by this fixture,\n            # but used as contextmanager, to setup and teardown configured ports.\n            root = TestEnvironmentAccess()\n    \n            # ensure create/release lifecycle of ports by TestEnvironmentAccess\n>           with root:\n\n.venv/lib/python3.11/site-packages/openta_pytest/plugin.py:73: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n.venv/lib/python3.11/site-packages/openta/testing/testenv.py:134: in __enter__\n    self.setup()\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <openta.testing.testenv.TestEnvironmentAccess object at 0x7cdd73ac6c50>\n\n    def setup(self) -> None:\n        try:\n            with allure.step(\"Setup Ports\"):\n                _logger.info(\"Setup Ports\")\n                for port in Registry().get_ports():\n                    port_repr = f\"[{port.order}] '{port.name}' {port.__class__.__name__}\"\n    \n                    if port.state < PortState.CREATED <= port.target_state:\n                        _logger.info(f\"Setup {port_repr}: Create\")\n                        port.create()\n    \n                    # configure\n                    if port.state < PortState.CONNECTED <= port.target_state:\n                        _logger.info(f\"Setup {port_repr}: Connect/Configure\")\n                        port.connect()\n    \n                    # and start if needed\n                    if port.state < PortState.STARTED <= port.target_state:\n                        _logger.info(f\"Setup {port_repr}: Start\")\n                        port.start()\n    \n        except Exception as err:\n            _logger.exception(f\"Setup Ports Failed: {exceptions.format_exc(err)}\")  # noqa: TRY401\n            # Initiate the Tear Down of already set up ports.\n            self.teardown()\n>           raise exceptions.PortError(f\"Setup of Ports Failed: {exceptions.format_exc(err)}\") from err\nE           openta.common.exceptions.PortError: Setup of Ports Failed: dSPACE.XIL.Testbench.ECUPort.Interfaces.Error.ECUPortException_impl: No valid license found. Vendor Description: CodeMeter License Server not found, Error 101.\nE              at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CheckAndThrowException(ServiceException exception)\nE              at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CreateInstance(InstanceType instanceType, Object[] args)\nE              at dSPACE.XIL.Testbench.Client.Testbench.ECUPort.ECUPortFactory.CreateECUPort(String name)\nE           ␤Failed to create 'ECU' DsEcuPort\n\n.venv/lib/python3.11/site-packages/openta/testing/testenv.py:178: PortError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":1,"retriesStatusChange":true,"beforeStages":[{"name":"ta","time":{"start":1756272692974,"stop":1756272708440,"duration":15466},"status":"broken","statusMessage":"openta.common.exceptions.PortError: Setup of Ports Failed: dSPACE.XIL.Testbench.ECUPort.Interfaces.Error.ECUPortException_impl: No valid license found. Vendor Description: CodeMeter License Server not found, Error 101.\n   at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CheckAndThrowException(ServiceException exception)\n   at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CreateInstance(InstanceType instanceType, Object[] args)\n   at dSPACE.XIL.Testbench.Client.Testbench.ECUPort.ECUPortFactory.CreateECUPort(String name)\n␤Failed to create 'ECU' DsEcuPort\n","statusTrace":"  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/pluggy/_callers.py\", line 38, in run_old_style_hookwrapper\n    res = yield\n          ^^^^^\n  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/_pytest/setuponly.py\", line 36, in pytest_fixture_setup\n    return (yield)\n            ^^^^^\n  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/pluggy/_callers.py\", line 121, in _multicall\n    res = hook_impl.function(*args)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/_pytest/fixtures.py\", line 1195, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/_pytest/fixtures.py\", line 922, in call_fixture_func\n    fixture_result = next(generator)\n                     ^^^^^^^^^^^^^^^\n  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/openta_pytest/plugin.py\", line 73, in ta\n    with root:\n  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/openta/testing/testenv.py\", line 134, in __enter__\n    self.setup()\n  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/openta/testing/testenv.py\", line 178, in setup\n    raise exceptions.PortError(f\"Setup of Ports Failed: {exceptions.format_exc(err)}\") from err\n","steps":[{"name":"Setup Ports","time":{"start":1756272693055,"stop":1756272708390,"duration":15335},"status":"broken","statusMessage":"dSPACE.XIL.Testbench.ECUPort.Interfaces.Error.ECUPortException_impl: No valid license found. Vendor Description: CodeMeter License Server not found, Error 101.\n   at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CheckAndThrowException(ServiceException exception)\n   at dSPACE.XIL.Testbench.Client.CommonServiceAccess.CreateInstance(InstanceType instanceType, Object[] args)\n   at dSPACE.XIL.Testbench.Client.Testbench.ECUPort.ECUPortFactory.CreateECUPort(String name)\n\nFailed to create 'ECU' DsEcuPort\n","statusTrace":"  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/openta/testing/testenv.py\", line 162, in setup\n    port.create()\n  File \"/home/runner/actions-runner/_work/dspace-virtual-workbench/dspace-virtual-workbench/PythonSDK_Turnlamp/.venv/lib/python3.11/site-packages/openta/ports/xilapi_dspace_ecu.py\", line 100, in create\n    self._port = ext_int.IDSTestbench(self._testbench).ECUPortFactory.CreateECUPort(self.name)\n                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"Teardown Ports","time":{"start":1756272708392,"stop":1756272708439,"duration":47},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":2,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"afterStages":[],"labels":[{"name":"tag","value":"turnlamp"},{"name":"parentSuite","value":"tests.turnlamp"},{"name":"suite","value":"test_synchron_flash"},{"name":"host","value":"ip-10-2-4-185"},{"name":"thread","value":"3207-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.turnlamp.test_synchron_flash"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"battery_voltage","value":"8"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[{"uid":"8554188a111c03ad","status":"passed","time":{"start":1756272703178,"stop":1756272706486,"duration":3308}}],"categories":[{"name":"Test defects","matchedStatuses":[]}],"tags":["turnlamp"]},"source":"1c0332ae3f13d75e.json","parameterValues":["8"]}